kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluator.java

```diff
@@ -589,6 +605,56 @@ ColumnVector visitCoalesce(ScalarExpression coalesce) {
           });
     }
 
+    @Override
+    ColumnVector visitAdd(ScalarExpression add) {
+      List<ColumnVector> childResults =
+          add.getChildren().stream().map(this::visit).collect(toList());
+      if (childResults.size() != 2) {
+        throw unsupportedExpressionException(
+            add, "ADD requires exactly two arguments: left and right operands");
+      }
+
+      return DefaultExpressionUtils.arithmeticVector(
+          childResults.get(0),
+          childResults.get(1),
+          new ArithmeticOperator() {
+            @Override
+            public byte apply(byte a, byte b) {
+              return (byte) (a + b);
+            }
+
+            @Override
+            public short apply(short a, short b) {
+              return (short) (a + b);
+            }
+
+            @Override
+            public int apply(int a, int b) {
+              return a + b;
+            }
+
+            @Override
+            public long apply(long a, long b) {
+              return a + b;
+            }
+
+            @Override
+            public float apply(float a, float b) {
+              return a + b;
+            }
+
+            @Override
+            public double apply(double a, double b) {
+              return a + b;
+            }
+
+            @Override
+            public BigDecimal apply(BigDecimal a, BigDecimal b) {
+              return a.add(b);
+            }
```

Comment:
I would leave decimal out for now, we don't need it and we're setting a precedent for behavior that is most likely not what a sane query engine would expect. E.g. the result may not have the same precision and scale than the two operands, we may decide to increase the precision, or lose some of the scale from one of the operand in some cases.